sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/ListingFileCatalog.scala

```diff
@@ -66,30 +66,13 @@ class ListingFileCatalog(
   }
 
   override def refresh(): Unit = {
-    val files = listLeafFiles(paths)
+    val files = HadoopFsRelation.listLeafFilesInParallel(paths, hadoopConf, sparkSession)
     cachedLeafFiles =
       new mutable.LinkedHashMap[Path, FileStatus]() ++= files.map(f => f.getPath -> f)
     cachedLeafDirToChildrenFiles = files.toArray.groupBy(_.getPath.getParent)
     cachedPartitionSpec = null
   }
 
-  protected def listLeafFiles(paths: Seq[Path]): mutable.LinkedHashSet[FileStatus] = {
-    if (paths.length >= sparkSession.sessionState.conf.parallelPartitionDiscoveryThreshold) {
-      HadoopFsRelation.listLeafFilesInParallel(paths, hadoopConf, sparkSession.sparkContext)
```

Comment:
Sorry. Let's keep the current behavior for now. It will be good to have a very small PR for this change. So we can try to get it in branch 2.0.